sky/spot/spot_state.py

```diff
@@ -308,79 +377,119 @@ def set_cancelled(job_id: int):
          SpotStatus.CANCELLING.value))
     _CONN.commit()
     logger.info('Job cancelled.')
 
 
 # ======== utility functions ========
-
-
 def get_nonterminal_job_ids_by_name(name: Optional[str]) -> List[int]:
     """Get non-terminal job ids by name."""
-    name_filter = 'AND job_name=(?)' if name is not None else ''
     field_values = [status.value for status in SpotStatus.terminal_statuses()]
+
+    name_filter = ''
     if name is not None:
-        field_values.append(name)
+        name_filter = 'AND (spot.job_name=(?) OR job_names.name=(?))'
+        field_values.extend([name, name])
+
     statuses = ', '.join(['?'] * len(SpotStatus.terminal_statuses()))
     rows = _CURSOR.execute(
         f"""\
-        SELECT job_id FROM spot
+        SELECT DISTINCT spot.new_job_id
+        FROM spot
+        LEFT OUTER JOIN job_names
+        ON spot.new_job_id=job_names.new_job_id
         WHERE status NOT IN
         ({statuses})
-        {name_filter}""", field_values)
+        {name_filter}""", field_values).fetchall()
     job_ids = [row[0] for row in rows if row[0] is not None]
     return job_ids
 
 
-def get_status(job_id: int) -> Optional[SpotStatus]:
-    """Get the status of a job."""
-    status = _CURSOR.execute(
+def _get_all_task_ids_statuses(job_id: int) -> List[Tuple[int, SpotStatus]]:
+    id_statuses = _CURSOR.execute(
         """\
-        SELECT status FROM spot WHERE job_id=(?)""", (job_id,)).fetchone()
-    if status is None:
-        return None
-    return SpotStatus(status[0])
+        SELECT task_id, status FROM spot
+        WHERE new_job_id=(?)
+        ORDER BY task_id ASC""", (job_id,)).fetchall()
+    return id_statuses
+
+
+def get_num_tasks(job_id: int) -> int:
+    return len(_get_all_task_ids_statuses(job_id))
+
+
+def get_latest_task_id_status(
+        job_id: int) -> Union[Tuple[int, SpotStatus], Tuple[None, None]]:
+    """Get the latest sub job id and status of a job."""
+    id_statuses = _get_all_task_ids_statuses(job_id)
+    if len(id_statuses) == 0:
+        return None, None
+    task_id, status = id_statuses[-1]
+    status = SpotStatus(status)
+    for task_id, status in id_statuses:
+        status = SpotStatus(status)
+        if not status.is_terminal():
+            break
+    return task_id, status
+
+
+def get_status(job_id: int) -> Optional[SpotStatus]:
+    _, status = get_latest_task_id_status(job_id)
+    return status
 
 
 def get_failure_reason(job_id: int) -> Optional[str]:
     """Get the failure reason of a job."""
     reason = _CURSOR.execute(
         """\
-        SELECT failure_reason FROM spot WHERE job_id=(?)""",
-        (job_id,)).fetchone()
-    if reason is None:
+        SELECT failure_reason FROM spot WHERE new_job_id=(?)""",
+        (job_id,)).fetchall()
+    reason = [r[0] for r in reason if r[0]]
+    if len(reason) == 0:
         return None
-    # reason[0] will be None if it is unfilled.
     return reason[0]
```

Comment:
docstr on something like returns the first failure reason in a spot job (chain)?